### PR TITLE
Issue #1402 Code refactoring around UpdateProfileConstants

### LINKED ISSUE #1402: Refactor code of function UpdateProfileConstants
In function `UpdateProfileConstants()` we change some variables from `pkg/minikube/contstants/constants.go` [1] . It is because cluster.GetHostStatus(api) uses `constants.MachineName` directly.

However we should not change these outside of `cmd/minishift/cmd/root.go`. We should find out how we can handle this in a better way.  

```
func UpdateProfileConstants(profileName string) {
	constants.ProfileName = profileName
	constants.MachineName = constants.ProfileName
	constants.Minipath = constants.GetProfileHomeDir()
}
```
[1] https://github.com/minishift/minishift/blob/master/pkg/minishift/profile/profile.go#L74-L83

### PR DESCRIPTION
**Pull request by LalatenduMohanty**
Signed-off-by: Lalatendu Mohanty <lmohanty@redhat.com>

Fixes #1402

### CODE REVIEW

**Comment by LalatenduMohanty** [comment]
CI has passed on this PR https://ci.centos.org/job/minishift-pr/5159/ . The push is only to update the commit message as @gbraad suggested.

File: cmd/minishift/cmd/profile/profile_delete.go [hunk 1]
```diff
@@ -69,7 +67,8 @@ func runProfileDelete(cmd *cobra.Command, args []string) {
 		}
 	}
 
-	api := libmachine.NewClient(profileBaseDir, constants.MakeMiniPath("certs"))
+	profileDirs := cmdState.NewMinishiftDirs(constants.GetProfileHomeDir(profileName))
```
Comment (praveenkumar) [inline cmd/minishift/cmd/profile/profile_delete.go]: @LalatenduMohanty `NewMinishiftDirs` is something reminds that we are doing a contractor operation here like creating those directories but in reality, we are just getting the respective directory Name under the profile. I think we should call it as `GetMinishiftDirs` instead `NewMinishiftDirs`

File: cmd/minishift/cmd/util/oc.go [hunk 2]
```diff
@@ -63,12 +63,11 @@ func CacheOc(openShiftVersion string) string {
...
 
 func SetOcContext(profileName string) error {
-	profileActions.UpdateProfileConstants(profileName)
-
 	// Need to create the kube config path for the profile for ocrunner to use it.
-	kubeConfigPath := filepath.Join(constants.Minipath, "machines", constants.MachineName+"_kubeconfig")
+	kubeConfigPath := filepath.Join(constants.Minipath, "machines", profileName+"_kubeconfig")
 
-	api := libmachine.NewClient(constants.Minipath, constants.MakeMiniPath("certs"))
+	profileDirs := cmdState.NewMinishiftDirs(constants.GetProfileHomeDir(profileName))
```
Comment (praveenkumar) [inline cmd/minishift/cmd/util/oc.go]: same here.

**Comment by gbraad** [comment]
retest this please

File: cmd/minishift/cmd/profile/profile_delete.go (hunk 1 above)
Comment (praveenkumar) [reply cmd/minishift/cmd/profile/profile_delete.go]: @LalatenduMohanty Sure that would do also.

**Comment by LalatenduMohanty** [comment]
retest this please

File: cmd/minishift/cmd/config/set_test.go [hunk 3]
```diff
@@ -35,7 +35,7 @@ func TestNotFound(t *testing.T) {
 func TestModifyData(t *testing.T) {
 	testDir, err := ioutil.TempDir("", "minishift-config-")
 	assert.NoError(t, err, "Error creating temp directory")
-	state.InstanceDirs = state.NewMinishiftDirs(testDir)
+	state.InstanceDirs = state.GetMinishiftDirsStructure(testDir)
```
Comment (praveenkumar) [inline cmd/minishift/cmd/config/set_test.go]: 👍

File: cmd/minishift/cmd/util/util.go [hunk 4]
```diff
@@ -77,7 +77,7 @@ func ExitIfNotRunning(driver drivers.Driver, machineName string) {
 
 func GetVMStatus(profileName string) string {
 	var status string
-	profileDirs := cmdState.NewMinishiftDirs(constants.GetProfileHomeDir(profileName))
+	profileDirs := cmdState.GetMinishiftDirsStructure(constants.GetProfileHomeDir(profileName))
```
Comment (praveenkumar) [inline cmd/minishift/cmd/util/util.go]: 👍

File: pkg/minishift/profile/profile.go [hunk 5]
```diff
@@ -80,17 +80,6 @@ func GetActiveProfile() string {
...
-// This needs a better solution than this as these variables should not be
-// changed outside of cmd/root.go. However cluster.GetHostStatus(api) uses
-// constants.MachineName inside the function.
-// This is a temporary fix and we will findout a better way to do it.
-func UpdateProfileConstants(profileName string) {
-	constants.ProfileName = profileName
-	constants.MachineName = constants.ProfileName
-	constants.Minipath = constants.GetProfileHomeDir(constants.ProfileName)
-}
-
```
Comment (praveenkumar) [inline pkg/minishift/profile/profile.go]: 👍

**Pull request merged**